src/main/java/org/smartregister/pathevaluator/action/ActionHelper.java

```diff
@@ -122,9 +122,9 @@ public static ResourceType getResourceType(Action action) {
 								"$this.item.where(linkId='locationId' and definition='details').answer.value.value");
 
 				if (locationIdStringValue != null) {
-					return stockDao.findInventoryInAServicePoint(locationIdStringValue.string());
+					return stockDao.findInventoryItemsInAJurisdiction(locationIdStringValue.string());
```

Comment:
This will lead to tasks for all service points to be generated on closed when any event from any service point is submitted.
`findInventoryInAServicePoint` should still be used however the expression `FHIRPathStringValue locationIdStringValue` should be pointing to the service point and not the jurisdiction

Reply:
actually use questionnaireReponse.subject which will have the stockId. Use that got get the inventory. You can wrap it `Collections.singleTon`

Reply:
Does that mean, I should fetch Inventory details from the stock Id (which will be inside `questionnaireReponse.subject`), and no need to check about location id or service point id?

Reply:
yes that's correct

Reply:
I assume that stockId inside `questionnaireReponse.subject` should be a numeric id (the primary key of the stock).
Is this correct @githengi ?

Reply:
As per discussion, it should be `_id`